bbconf/modules/bedfiles.py

```diff
@@ -0,0 +1,726 @@
+        )
+        return None
+
+    def text_to_bed_search(
+        self, query: str, limit: int = 10, offset: int = 0
+    ) -> BedListSearchResult:
+        """
+        Search for bed files by text query in qdrant database
+
+        :param query: text query
+        :param limit: number of results to return
+        :param offset: offset to start from
+
+        :return: list of bed file metadata
+        """
+        _LOGGER.info(f"Looking for: {query}")
+        _LOGGER.info(f"Using backend: {self._config.t2bsi}")
+
+        results = self._config.t2bsi.nl_vec_search(query, limit=limit, offset=offset)
+        results_list = []
+        for result in results:
+            result_id = result["id"].replace("-", "")
+            try:
+                result_meta = self.get(result_id)
+            except BEDFileNotFoundError as e:
+                _LOGGER.warning(
+                    f"Could not retrieve metadata for bed file: {result_id}. Error: {e}"
+                )
+                continue
+            if result_meta:
+                results_list.append(QdrantSearchResult(**result, metadata=result_meta))
+        return BedListSearchResult(
+            count=len(results), limit=limit, offset=offset, results=results_list
+        )
+
+    def reindex_qdrant(self) -> None:
+        """
+        Re-upload all files to quadrant.
+        !Warning: only hg38 genome can be added to qdrant!
+
+        If you want want to fully reindex/reupload to qdrant, first delete collection and create new one.
+
+        Upload all files to qdrant.
+        """
+        bb_client = BBClient()
+
+        statement = select(Bed.id).where(Bed.genome_alias == QDRANT_GENOME)
+
+        with Session(self._db_engine.engine) as session:
+            bed_ids = session.execute(statement).all()
+
+        bed_ids = [bed_result[0] for bed_result in bed_ids]
+
+        for record_id in bed_ids:
+            bed_region_set_obj = bb_client.load_bed(record_id)
+
+            self.upload_file_qdrant(
+                bed_id=record_id,
+                bed_file=bed_region_set_obj,
+                payload={"bed_id": record_id},
+            )
+
+        return None
+
+    def delete_qdrant_point(self, identifier: str) -> None:
+        """
+        Delete bed file from qdrant.
+
+        :param identifier: bed file identifier
+        :return: None
+        """
+
+        result = self._config.qdrant_engine.qd_client.delete(
+            collection_name=self._config.config.qdrant.collection,
+            points_selector=PointIdsList(
+                points=[identifier],
+            ),
+        )
+        return result
+
+    def create_qdrant_collection(self) -> bool:
+        """
+        Create qdrant collection for bed files.
+        """
+        return self._config.qdrant_engine.qd_client.create_collection(
+            collection_name=self._config.config.qdrant.collection,
+            vectors_config=VectorParams(size=100, distance=Distance.DOT),
```

Comment:
Can we parametrize the embedding size? I believe that the models should be able to spit out an embedding dimension for you